### PR TITLE
Adding support for handling mac address

### PR DESCRIPTION
**Pull request by PavamanSubramaniyam**
supporting  mac address which is being provided as interface parameter value in yaml files

### CODE REVIEW

**Comment by PavamanSubramaniyam** [comment]
cat udaddy.py.data/udaddy_roce.yaml
Test: !mux
    options: !mux
        first:
            basic_option: ' '
        second:
            basic_option: -c 50 -C 30 -S 1024
        third:
            basic_option: -t tcp
        fourth:
            ext_option: -f ip
        fifth:
            ext_option: -f ipv6
        sixth:
            ext_option: -P udp
parameters:
    ext_flag: "1"
    interface: "16:77:03:4b:8f:03"
    peer_ip: 17.1.1.183
    peer_user: "root"
    peer_password: "llm@fvt"
    host_ip: "17.1.1.23"
    netmask: "255.255.255.0"
mtu: !mux
    1500:
        mtu: "1500"
    2000:
        mtu: "2000"
    3000:
        mtu: "3000"
    4000:
        mtu: "4000"
    5000:
        mtu: "5000"
    6000:
        mtu: "6000"
    7000:
        mtu: "7000"
    8000:
        mtu: "8000"
    9000:
        mtu: "9000"

avocado run --max-parallel-tasks=1 udaddy.py -m udaddy.py.data/udaddy_roce.yaml                                                                              JOB ID     : 1710333df091d9cbef607a4d68c1e12b6790db45
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-02-25T02.34-1710333/job.log
 (01/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-1500-parameters-258a: STARTED
 (01/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-1500-parameters-258a: PASS (10.84 s)
 (02/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-2000-parameters-6668: STARTED
 (02/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-2000-parameters-6668: PASS (10.72 s)
 (03/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-3000-parameters-72e1: STARTED
 (03/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-3000-parameters-72e1: PASS (10.72 s)
 (04/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-4000-parameters-25f8: STARTED
 (04/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-4000-parameters-25f8: PASS (10.72 s)
 (05/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-5000-parameters-7652: STARTED
 (05/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-5000-parameters-7652: PASS (10.71 s)
 (06/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-6000-parameters-6e79: STARTED
 (06/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-6000-parameters-6e79: PASS (10.73 s)
 (07/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-7000-parameters-1072: STARTED
 (07/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-7000-parameters-1072: PASS (10.68 s)
 (08/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-8000-parameters-b40d: STARTED
 (08/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-8000-parameters-b40d: PASS (10.73 s)
 (09/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-9000-parameters-4544: STARTED
 (09/54) udaddy.py:Udaddy.test;run-Test-options-first-mtu-9000-parameters-4544: PASS (10.73 s)
 (10/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-1500-parameters-820f: STARTED
 (10/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-1500-parameters-820f: PASS (10.62 s)
 (11/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-2000-parameters-03ea: STARTED
 (11/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-2000-parameters-03ea: PASS (10.85 s)
 (12/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-3000-parameters-2918: STARTED
 (12/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-3000-parameters-2918: PASS (10.87 s)
 (13/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-4000-parameters-a067: STARTED
 (13/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-4000-parameters-a067: PASS (10.87 s)
 (14/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-5000-parameters-e9f6: STARTED
 (14/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-5000-parameters-e9f6: PASS (10.86 s)
 (15/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-6000-parameters-42b5: STARTED
 (15/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-6000-parameters-42b5: PASS (10.85 s)
 (16/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-7000-parameters-546f: STARTED
 (16/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-7000-parameters-546f: PASS (10.86 s)
 (17/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-8000-parameters-881b: STARTED
 (17/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-8000-parameters-881b: PASS (10.88 s)
 (18/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-9000-parameters-1ecd: STARTED
 (18/54) udaddy.py:Udaddy.test;run-Test-options-second-mtu-9000-parameters-1ecd: PASS (10.87 s)
 (19/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-1500-parameters-1a12: STARTED
 (19/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-1500-parameters-1a12: PASS (10.48 s)
 (20/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-2000-parameters-f1ef: STARTED
 (20/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-2000-parameters-f1ef: PASS (10.73 s)
 (21/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-3000-parameters-191c: STARTED
 (21/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-3000-parameters-191c: PASS (10.73 s)
 (22/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-4000-parameters-ee47: STARTED
 (22/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-4000-parameters-ee47: PASS (10.69 s)
 (23/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-5000-parameters-58e5: STARTED
 (23/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-5000-parameters-58e5: PASS (10.71 s)
 (24/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-6000-parameters-4605: STARTED
 (24/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-6000-parameters-4605: PASS (10.72 s)
 (25/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-7000-parameters-d184: STARTED
 (25/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-7000-parameters-d184: PASS (10.73 s)
 (26/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-8000-parameters-3572: STARTED
 (26/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-8000-parameters-3572: PASS (10.74 s)
 (27/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-9000-parameters-1da3: STARTED
 (27/54) udaddy.py:Udaddy.test;run-Test-options-third-mtu-9000-parameters-1da3: PASS (10.73 s)
 (28/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-1500-parameters-00c5: STARTED
 (28/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-1500-parameters-00c5: PASS (10.46 s)
 (29/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-2000-parameters-49c0: STARTED
 (29/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-2000-parameters-49c0: PASS (10.72 s)
 (30/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-3000-parameters-044a: STARTED
 (30/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-3000-parameters-044a: PASS (10.73 s)
 (31/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-4000-parameters-9d3c: STARTED
 (31/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-4000-parameters-9d3c: PASS (10.68 s)
 (32/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-5000-parameters-ca7d: STARTED
 (32/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-5000-parameters-ca7d: PASS (10.73 s)
 (33/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-6000-parameters-eb40: STARTED
 (33/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-6000-parameters-eb40: PASS (10.72 s)
 (34/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-7000-parameters-206b: STARTED
 (34/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-7000-parameters-206b: PASS (10.72 s)
 (35/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-8000-parameters-90d7: STARTED
 (35/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-8000-parameters-90d7: PASS (10.72 s)
 (36/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-9000-parameters-bdf2: STARTED
 (36/54) udaddy.py:Udaddy.test;run-Test-options-fourth-mtu-9000-parameters-bdf2: PASS (10.75 s)
 (37/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-1500-parameters-e27d: STARTED
 (37/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-1500-parameters-e27d: PASS (10.46 s)
 (38/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-2000-parameters-63dc: STARTED
 (38/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-2000-parameters-63dc: PASS (10.72 s)
 (39/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-3000-parameters-a110: STARTED
 (39/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-3000-parameters-a110: PASS (10.72 s)
 (40/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-4000-parameters-68d6: STARTED
 (40/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-4000-parameters-68d6: PASS (10.73 s)
 (41/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-5000-parameters-eea2: STARTED
 (41/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-5000-parameters-eea2: PASS (10.72 s)
 (42/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-6000-parameters-b7ee: STARTED
 (42/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-6000-parameters-b7ee: PASS (10.71 s)
 (43/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-7000-parameters-1a8d: STARTED
 (43/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-7000-parameters-1a8d: PASS (10.72 s)
 (44/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-8000-parameters-489b: STARTED
 (44/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-8000-parameters-489b: PASS (10.74 s)
 (45/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-9000-parameters-ccce: STARTED
 (45/54) udaddy.py:Udaddy.test;run-Test-options-fifth-mtu-9000-parameters-ccce: PASS (10.73 s)
 (46/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-1500-parameters-d794: STARTED
 (46/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-1500-parameters-d794: PASS (10.46 s)
 (47/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-2000-parameters-d942: STARTED
 (47/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-2000-parameters-d942: PASS (10.72 s)
 (48/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-3000-parameters-b579: STARTED
 (48/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-3000-parameters-b579: PASS (10.72 s)
 (49/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-4000-parameters-7ee5: STARTED
 (49/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-4000-parameters-7ee5: PASS (10.71 s)
 (50/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-5000-parameters-e990: STARTED
 (50/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-5000-parameters-e990: PASS (10.71 s)
 (51/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-6000-parameters-6b24: STARTED
 (51/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-6000-parameters-6b24: PASS (10.72 s)
 (52/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-7000-parameters-e57f: STARTED
 (52/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-7000-parameters-e57f: PASS (10.72 s)
 (53/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-8000-parameters-a16a: STARTED
 (53/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-8000-parameters-a16a: PASS (10.73 s)
 (54/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-9000-parameters-4e96: STARTED
 (54/54) udaddy.py:Udaddy.test;run-Test-options-sixth-mtu-9000-parameters-4e96: PASS (10.72 s)
RESULTS    : PASS 54 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-02-25T02.34-1710333/results.html
JOB TIME   : 720.93 s

File: io/net/infiniband/ib_pingpong.py [hunk 1]
```diff
@@ -44,16 +44,24 @@ def setUp(self):
...
         '''
+        local = LocalHost()
         interfaces = netifaces.interfaces()
+        device = self.params.get("interface", default=None)
+        if device in interfaces:
+            self.iface = device
+        elif local.validate_mac_addr(device) and device in local.get_all_hwaddr():
+            self.iface = local.get_interface_by_hwaddr(device).name
+        else:
+            self.iface = None
```
Comment (Naresh-ibm) [inline io/net/infiniband/ib_pingpong.py:55]: @PavamanSubramaniyam I think we don't need this as we are cancelling the test any way.

File: io/net/infiniband/mckey.py [hunk 2]
```diff
@@ -63,15 +63,24 @@ def setUp(self):
...
+        local = LocalHost()
         interfaces = netifaces.interfaces()
-        self.iface = self.params.get("interface", default="")
+        device = self.params.get("interface", default=None)
+        if device in interfaces:
+            self.iface = device
+        elif local.validate_mac_addr(device) and device in local.get_all_hwaddr():
+            self.iface = local.get_interface_by_hwaddr(device).name
+        else:
+            self.iface = None
```
Comment (Naresh-ibm) [inline io/net/infiniband/mckey.py:75]: same here and below other places.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]
Also please attach the runs logs for all the tests. I could see only for udaddy

File: io/net/infiniband/ib_pingpong.py (hunk 1 above)
Comment (abdhaleegit) [reply io/net/infiniband/ib_pingpong.py:55]: @Naresh-ibm we need this for teardown.. else the cancel fails abrubtly

**Comment by abdhaleegit** [comment]
@PavamanSubramaniyam  handle the teardown incase of invalid interface  i.e self.iface = None  make sure teardown checks for this and move all successding teardown operation inside the if self.iface check

**Comment by PavamanSubramaniyam** [comment]
I have added the if self.iface check in teardown and pushed the changes again

File: io/net/infiniband/ib_pingpong.py (hunk 1 above)
Comment (abdhaleegit) [reply io/net/infiniband/ib_pingpong.py:55]: thanks

**Comment by abdhaleegit** [comment]
@PavamanSubramaniyam one quick run before I merge

**Comment by PavamanSubramaniyam** [comment]
avocado run --max-parallel-tasks=1 ib_pingpong.py -m ib_pingpong.py.data/ib_pingpong_roce.yaml
JOB ID     : 80e87d8b10c38840d56a7aa3efcaf6656e7b9f37
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-01-15T06.40-80e87d8/job.log
 (01/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-1500-parameters-4714: STARTED
 (01/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-1500-parameters-4714: PASS (27.47 s)
 (02/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-2000-parameters-a912: STARTED
 (02/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-2000-parameters-a912: PASS (27.41 s)
 (03/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-3000-parameters-d5c7: STARTED
 (03/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-3000-parameters-d5c7: PASS (27.33 s)
 (04/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-4000-parameters-aabc: STARTED
 (04/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-4000-parameters-aabc: PASS (27.36 s)
 (05/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-5000-parameters-c509: STARTED
 (05/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-5000-parameters-c509: PASS (27.34 s)
 (06/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-6000-parameters-d575: STARTED
 (06/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-6000-parameters-d575: PASS (27.35 s)
 (07/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-7000-parameters-baa8: STARTED
 (07/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-7000-parameters-baa8: PASS (27.36 s)
 (08/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-8000-parameters-ecda: STARTED
 (08/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-8000-parameters-ecda: PASS (27.33 s)
 (09/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-9000-parameters-ab29: STARTED
 (09/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-ud_pingpong-mtu-9000-parameters-ab29: PASS (27.32 s)
 (10/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-1500-parameters-9a7e: STARTED
 (10/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-1500-parameters-9a7e: PASS (46.80 s)
 (11/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-2000-parameters-ca43: STARTED
 (11/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-2000-parameters-ca43: PASS (47.51 s)
 (12/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-3000-parameters-4274: STARTED
 (12/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-3000-parameters-4274: PASS (47.78 s)
 (13/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-4000-parameters-9794: STARTED
 (13/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-4000-parameters-9794: PASS (47.54 s)
 (14/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-5000-parameters-f9da: STARTED
 (14/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-5000-parameters-f9da: PASS (47.72 s)
 (15/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-6000-parameters-0b4b: STARTED
 (15/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-6000-parameters-0b4b: PASS (47.51 s)
 (16/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-7000-parameters-bc73: STARTED
 (16/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-7000-parameters-bc73: PASS (47.54 s)
 (17/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-8000-parameters-cccb: STARTED
 (17/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-8000-parameters-cccb: PASS (47.49 s)
 (18/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-9000-parameters-90cd: STARTED
 (18/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-uc_pingpong-mtu-9000-parameters-90cd: PASS (47.65 s)
 (19/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-1500-parameters-1c24: STARTED
 (19/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-1500-parameters-1c24: PASS (46.82 s)
 (20/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-2000-parameters-218a: STARTED
 (20/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-2000-parameters-218a: PASS (47.51 s)
 (21/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-3000-parameters-b962: STARTED
 (21/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-3000-parameters-b962: PASS (47.57 s)
 (22/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-4000-parameters-c550: STARTED
 (22/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-4000-parameters-c550: PASS (47.76 s)
 (23/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-5000-parameters-b597: STARTED
 (23/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-5000-parameters-b597: PASS (47.58 s)
 (24/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-6000-parameters-a5f7: STARTED
 (24/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-6000-parameters-a5f7: PASS (47.75 s)
 (25/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-7000-parameters-964b: STARTED
 (25/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-7000-parameters-964b: PASS (47.68 s)
 (26/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-8000-parameters-9689: STARTED
 (26/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-8000-parameters-9689: PASS (47.66 s)
 (27/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-9000-parameters-e62c: STARTED
 (27/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-rc_pingpong-mtu-9000-parameters-e62c: PASS (47.74 s)
 (28/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-1500-parameters-b22f: STARTED
 (28/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-1500-parameters-b22f: PASS (46.94 s)
 (29/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-2000-parameters-71e5: STARTED
 (29/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-2000-parameters-71e5: PASS (46.96 s)
 (30/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-3000-parameters-5c1c: STARTED
 (30/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-3000-parameters-5c1c: PASS (47.03 s)
 (31/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-4000-parameters-1748: STARTED
 (31/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-4000-parameters-1748: PASS (46.69 s)
 (32/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-5000-parameters-7c2d: STARTED
 (32/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-5000-parameters-7c2d: PASS (46.71 s)
 (33/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-6000-parameters-4196: STARTED
 (33/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-6000-parameters-4196: PASS (47.31 s)
 (34/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-7000-parameters-22a5: STARTED
 (34/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-7000-parameters-22a5: PASS (46.61 s)
 (35/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-8000-parameters-17de: STARTED
 (35/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-8000-parameters-17de: PASS (46.75 s)
 (36/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-9000-parameters-5ccf: STARTED
 (36/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-srq_pingpong-mtu-9000-parameters-5ccf: PASS (46.86 s)
 (37/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-1500-parameters-cbb1: STARTED
 (37/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-1500-parameters-cbb1: PASS (34.33 s)
 (38/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-2000-parameters-c050: STARTED
 (38/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-2000-parameters-c050: PASS (35.01 s)
 (39/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-3000-parameters-8fb5: STARTED
 (39/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-3000-parameters-8fb5: PASS (34.87 s)
 (40/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-4000-parameters-0387: STARTED
 (40/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-4000-parameters-0387: PASS (35.05 s)
 (41/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-5000-parameters-ac74: STARTED
 (41/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-5000-parameters-ac74: PASS (34.91 s)
 (42/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-6000-parameters-af13: STARTED
 (42/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-6000-parameters-af13: PASS (35.00 s)
 (43/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-7000-parameters-ed05: STARTED
 (43/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-7000-parameters-ed05: PASS (34.99 s)
 (44/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-8000-parameters-8624: STARTED
 (44/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-8000-parameters-8624: PASS (34.87 s)
 (45/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-9000-parameters-d453: STARTED
 (45/45) ib_pingpong.py:PingPong.test_ib_pingpong;run-Test-xsrq_pingpong-mtu-9000-parameters-d453: PASS (35.04 s)
RESULTS    : PASS 45 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by PavamanSubramaniyam** [comment]
I have pasted the complete results in the Jira task for all the ROCE tests for which the code changes are made. 
I am pasting the execution results for only couple of tests here after the review comments are incorporated and resent the changes.

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Pull request merged**